### PR TITLE
Add connect Wallet modal

### PR DESCRIPTION
**Pull request by Nazeeh21**
Added a wallet connect modal, `AddressContext` and modified the type of context in `KitProvider` context

Things remained: Disconnect the wallet by calling a certain method so that even if the user reloads, it does not connect the wallet unless the user manually connects again

### CODE REVIEW

File: package/src/components/Wallets/walletConnectWallet.tsx [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+  useSetWalletProvider,
+} from '../KitProvider/AddressContext';
+import { useSetWalletClient } from '../KitProvider/KitProvider';
+import {
+  useCurrentChain,
+  useSupportedChains,
+} from '../KitProvider/ChainContext';
+import { storePrevAccount, storePrevWallet } from '../../utils/utils';
+
+const projectId = '5a13f1a5297da2cd768519079890e4fe';
```
Comment (miralsuthar) [inline package/src/components/Wallets/walletConnectWallet.tsx:17]: ask for project id in setup.

File: tsconfig.json [hunk 2]
```diff
@@ -73,7 +73,8 @@
     "package/**/*.ts",
     ".eslintrc.js",
     "jest.config.js",
-    "stories/**.tsx"
+    "stories/**.tsx",
+    "package/src/custom.d.ts"
```
Comment (miralsuthar) [inline tsconfig.json:77]: remove this cause we are not using this atm.

**Review by miralsuthar — COMMENTED** [review]
looks good, just some changes mentioned in comment

File: tsconfig.json (hunk 2 above)
Comment (Nazeeh21) [reply tsconfig.json:77]: Removed

**Review by miralsuthar — COMMENTED** [review]
Good for merge

**Review by miralsuthar — APPROVED** [review]
Good for merge

**Pull request merged**